sympy/physics/mechanics/inertia.py

```diff
@@ -0,0 +1,195 @@
+from sympy.core.backend import sympify
+from sympy.physics.vector import Point, Dyadic, ReferenceFrame
+
+__all__ = ['inertia', 'inertia_of_point_mass', 'Inertia']
+
+
+def inertia(frame, ixx, iyy, izz, ixy=0, iyz=0, izx=0):
+    """Simple way to create inertia Dyadic object.
+
+    Explanation
+    ===========
+
+    If you do not know what a Dyadic is, just treat this like the inertia
+    tensor. Then, do the easy thing and define it in a body-fixed frame.
+
+    Parameters
+    ==========
+
+    frame : ReferenceFrame
+        The frame the inertia is defined in
+    ixx : Sympifyable
+        the xx element in the inertia dyadic
+    iyy : Sympifyable
+        the yy element in the inertia dyadic
+    izz : Sympifyable
+        the zz element in the inertia dyadic
+    ixy : Sympifyable
+        the xy element in the inertia dyadic
+    iyz : Sympifyable
+        the yz element in the inertia dyadic
+    izx : Sympifyable
+        the zx element in the inertia dyadic
+
+    Examples
+    ========
+
+    >>> from sympy.physics.mechanics import ReferenceFrame, inertia
+    >>> N = ReferenceFrame('N')
+    >>> inertia(N, 1, 2, 3)
+    (N.x|N.x) + 2*(N.y|N.y) + 3*(N.z|N.z)
+
+    """
+
+    if not isinstance(frame, ReferenceFrame):
+        raise TypeError('Need to define the inertia in a frame')
+    ixx = sympify(ixx)
+    ixy = sympify(ixy)
+    iyy = sympify(iyy)
+    iyz = sympify(iyz)
+    izx = sympify(izx)
+    izz = sympify(izz)
+    ol = ixx * (frame.x | frame.x)
+    ol += ixy * (frame.x | frame.y)
+    ol += izx * (frame.x | frame.z)
+    ol += ixy * (frame.y | frame.x)
+    ol += iyy * (frame.y | frame.y)
+    ol += iyz * (frame.y | frame.z)
+    ol += izx * (frame.z | frame.x)
+    ol += iyz * (frame.z | frame.y)
+    ol += izz * (frame.z | frame.z)
+    return ol
+
+
+def inertia_of_point_mass(mass, pos_vec, frame):
+    """Inertia dyadic of a point mass relative to point O.
+
+    Parameters
+    ==========
+
+    mass : Sympifyable
+        Mass of the point mass
+    pos_vec : Vector
+        Position from point O to point mass
+    frame : ReferenceFrame
+        Reference frame to express the dyadic in
+
+    Examples
+    ========
+
+    >>> from sympy import symbols
+    >>> from sympy.physics.mechanics import ReferenceFrame, inertia_of_point_mass
+    >>> N = ReferenceFrame('N')
+    >>> r, m = symbols('r m')
+    >>> px = r * N.x
+    >>> inertia_of_point_mass(m, px, N)
+    m*r**2*(N.y|N.y) + m*r**2*(N.z|N.z)
+
+    """
+
+    return mass * (((frame.x | frame.x) + (frame.y | frame.y) +
+                    (frame.z | frame.z)) * (pos_vec & pos_vec) -
+                   (pos_vec | pos_vec))
+
+
+class Inertia:
```

Comment:
Should we be subclassing `tuple` or `typing.NamedTuple` here in order to achieve correct typing behaviour for backwards compatibility? Inertias were previously tuples, if a user is using an `isinstance` check then this could fail with `Inertia`.

Reply:
I personally don't like inheriting from `tuple`, because we'll also get methods which we don't like (such as `__add__`). I can switch the design of this class to the first example proposed in [this comment](https://github.com/sympy/sympy/issues/24257#issue-1447827555), which will indeed ensure more complete backwards compatibility.

Reply:
I think inheriting from tuple would be the "proper" way to do it. However, I agree that getting `tuple`'s `__add__` is a negative so this would need to be overridden. I'd propose subclassing from `tuple` but then manually overwriting `__add__` and `__mul__` (and any other unwanted dunder methods) to raise `TypeError: + operator is not supported for Inertia` or something.

Reply:
The problem with a `dataclass` is that it still does not give the desired result where `isinstance(Inertia(...), tuple) == True`. Inheriting from a `collections.namedtuple`, like `Inertia(namedtuple('Inertia', ['dyadic', 'point']))` seems a bit weird and unnecessary. Therefore, I find `Inertia(tuple)` the most sensible, while also overwriting some magic methods like `__add__`, which would result it wrong behaviour.

Reply:
Okay, I see why you're doing this. Current code returns a tuple and you want to change it to an object with attributes and methods. In that case `namedtuple` is specifically designed for this situation. Yes, `__add__` might not do what you would ideally want but it will continue to function is it always did for anyone who might happen to be using it already. On the other hand if you actually want to change the behaviour of `__add__` to do something more useful then you can still override the `__add__` method.

Reply:
It is still not very recommended to override `namedtuple` for builtin methods like `__add__`, `__mul__` to have inconsistent behavior than `tuple` (such as vector addition), because you need to be careful you don't use that subclass where `tuple + tuple` is expected to give juxtaposition result, and it is easy to get this bleed through because `isinstance(..., tuple)` wouldn't guard it.
If we are unhappy with default like, we would better raise errors early or leave as is, and define other methods that doesn't exist in base class.

Reply:
Okay, I've reverted the override, while I have to say that I'm quite sure that most problems were guarded by also overwriting the `__radd__` and `__rmul__` methods. @sylee957 what did you by the way mean with _raise error early_?

Reply:
I don't have strong opinion about whether this shouldn't be subclassed or not, it can be subclassed if it was originally tuple, but only be careful about overriding `__add__` and `__mul__` to have different behavior, because that specific part is not compatible.

Reply:
In that case, I'll revert my last commit and block the usage of `__add__` and `__mul__`, because:
- They'll prevent the misusage of those operations, which is correct from a mechanical point of view.
- In the future it is easier to switch to supporting those operations in a way which is correct from an mechanical point of view. For example `5 * Inertia(dyadic, point) == Inertia(5 * dyadic, point)`.